delfin/drivers/netapp/dataontap/netapp_handler.py

```diff
@@ -949,3 +991,218 @@ def get_alert_sources(self):
+                              constants.ResourceType.STORAGE)
+            return storage_metrics
+
+    def get_pool_per(self, metrics, storage_id, start_time, end_time):
+        data = None
+        agg_info = self.ssh_do_exec(
+            constant.AGGREGATE_SHOW_DETAIL_COMMAND)
+        agg_map_list = []
+        pool_metrics = []
+        Tools.split_value_map_list(agg_info, agg_map_list, split=':')
+        for agg_map in agg_map_list:
+            if 'UUIDString' in agg_map:
+                uuid = agg_map['UUIDString']
+                json_info = self.do_rest_call(
+                    constant.POOL_PER_URL % uuid, data)
+                pool_metrics.extend(
+                    PerformanceHandler.get_per_value(
+                        metrics,
+                        storage_id,
+                        start_time,
+                        end_time,
+                        json_info,
+                        agg_map['UUIDString'],
+                        agg_map['Aggregate'],
+                        constants.ResourceType.STORAGE_POOL))
+        return pool_metrics
+
+    def get_volume_per(self, metrics, storage_id, start_time, end_time):
+        data = None
+        volume_info = \
+            self.ssh_do_exec(constant.LUN_SHOW_DETAIL_COMMAND)
+        volume_map_list = []
+        volume_metrics = []
+        Tools.split_value_map_list(volume_info, volume_map_list, split=':')
+        for volume in volume_map_list:
+            if 'LUNUUID' in volume:
+                uuid = volume['LUNUUID']
+                json_info = self.do_rest_call(
+                    constant.VOLUME_PER_URL % uuid, data)
+                volume_metrics.extend(
+                    PerformanceHandler.get_per_value(
+                        metrics, storage_id,
+                        start_time, end_time,
+                        json_info, volume['SerialNumber'],
+                        volume['LUNName'],
+                        constants.ResourceType.VOLUME))
+        return volume_metrics
+
+    def get_fs_per(self, metrics, storage_id, start_time, end_time):
+        fs_info = self.do_rest_call(
+            constant.FS_INFO_URL, {})
+        fs_metrics = []
+        for fs in fs_info:
+            if 'uuid' in fs:
+                data = None
+                uuid = fs['uuid']
+                json_info = self.do_rest_call(
+                    constant.FS_PER_URL % uuid, data)
+                fs_id = self.get_fs_id(
+                    fs['svm']['name'], fs['name'])
+                fs_metrics.extend(
+                    PerformanceHandler.get_per_value(
+                        metrics, storage_id, start_time,
+                        end_time, json_info, fs_id,
+                        fs['name'],
+                        constants.ResourceType.FILESYSTEM))
+        return fs_metrics
+
+    def get_port_per(self, metrics, storage_id, start_time, end_time):
+        fc_port = self.do_rest_call(constant.FC_INFO_URL, {})
```

Comment:
`{}` can be set to `None` , please check.

Reply:
Has been modified